noxfile.py

```diff
@@ -2,7 +2,7 @@
 
 import nox
 
-ALL_PYTHONS = ["3.6", "3.7", "3.8", "3.9", "3.10"]
+ALL_PYTHONS = ["3.7", "3.8", "3.9", "3.10"]
```

Comment:
Can we add 3.11 now, too? And 3.11-dev to the testing matrix? Awkward supports 3.11 now. Some of the dev dependencies will need to require <3.11.

Reply:
(Actually the 3.11 stuff can be added before 0.9 & this PR)